### ISSUE TITLE
Remove new opencmiss.iron in install and make install able to use PYTHONPATH for combined OpenCMISS

### ISSUE THREAD

**Issue by chrispbradley** [OpenCMISS/manage#89]
The install location for the python files has recently changed and now separates out the iron and zinc python bindings. This needs to be reverted as the whole point was to have a combined python bindings for OpenCMISS. The new install with separate directories now prevents setting the PYTHONPATH to the one directory containing the combined bindings. The install location needs to be able to be used when you are not using virtual environments.

**Comment by hsorby** [comment]
This is not possible to achieve, the way Python thinks about things the two packages are separate entities.  The PYTHONPATH can be set to both directories so that is not a problem.  

The suggested method is to use virtual environments using the PYTHONPATH is only a backup.

**Comment by hsorby** [comment]
Closing this issue as there is a solution to the problem raised.

**Comment by chrispbradley** [comment]
The solution is not to have opencmiss.iron. Is this the solution you are talking about?

**Comment by hsorby** [comment]
No, I said it is impossible for the the directory to be removed. The solution is to set the PYTHONPATH to include the directory if using the PYTHONPATH variable is what you have to use. 
 The best practice is to use virtual environments.

**Comment by chrispbradley** [comment]
You can keep closing this and I will keep reopening it. You have not said why the directory has been split. The current opencmiss.iron and opencmiss.zinc is against the agreed naming standard. What has been agreed is from opencmiss import iron and from opencmiss import zinc, not from opencmiss.iron. The idea is that it should be one opencmiss project. You changed this without discussion or agreement. opencmiss.iron is not an acceptable solution. What specific reason was opencmiss moved to opencmiss.iron?

**Comment by hsorby** [comment]
I think you have the wrong idea about what has happened.  The import statements have not changed they are as agreed.  As far as using OpenCMISS libraries from Python are concerned there is only one opencmiss package, it is a namespace package as far as Python is concerned.

If you insist on using the PYTHONPATH to access OpenCMISS libraries from Python you will need to append the /path/to/opencmiss.iron directory to the PYTHONPATH environment variable to access the iron library in the agreed manner `from opencmiss import iron`.

There is no combined bindings in this form, in this form they cannot exist in the same directory.  This form is not meant for use within Python directly using the PYTHONPATH is a workaround that allows this to happen.

**Comment by chrispbradley** [comment]
Please stop closing this. It is not sorted out. The bindings were in ..../opencmiss/iron and ..../opencmiss/zinc. They then moved to ..../opencmiss.iron/opencmiss/iron and ..../opencmiss.zinc/opencmiss/zinc. Why was this move done? What specifically stops both iron and zinc from being in the same directory tree under opencmiss? The plan _is_ to have combined bindings.

**Comment by hsorby** [comment]
The bindings *are* combined as far as Python is concerned it does not matter that they are not in the same physical directory not a jot does this matter.

The number one reason is that both packages must have the same file called 'setup.py'.

**Comment by chrispbradley** [comment]
Do not keep closing this issue. I will close it when there is a satisfactory resolution. 

Why would there be more than one setup.py file. There should only be one setup.py for the one opencmiss package?

**Comment by rchristie** [comment]
This is a total non-issue, a waste of everyone's time. We need to be able to independently build and install Iron and Zinc. Hugh has explained the technical reason why both Iron and Zinc need their own independent setup.py. Once *installed* into Python using the respective setup scripts, they appear in the same namespace, which is our only requirement. For those 0.1% of users who do not install the packages into Python in the default environment or in virtual environments you may still use PYTHONPATH with both folders in it, as Hugh explained above. Resolved.

**Comment by chrispbradley** [comment]
No, it is not a total non issue. Indeed it is a very important issue. It seems that the real issue is that two separate python packages are being setup and not one opencmiss package. This is why two setup.py s are needed and not one. We are developing one opencmiss package with an iron and zinc modules and not separate iron and zinc packages. When appearing on PyPI it needs to appear as opencmiss and not separate iron and zinc packages. There is absolutely no problem having iron and zinc to be able to be independently built and installed with one package and one setup.py. If you turn off iron or zinc then it will simply not build or install the component.

**Comment by rchristie** [comment]
The original request has been answered: Hugh has explained how PYTHONPATH can include both folders.

**Comment by chrispbradley** [comment]
The problem is not the PYTHONPATH but the separate packages.
